File: JobSearchingWebApp/Endpoints/CV/Update/CVUpdateRequest.cs
using JobSearchingWebApp.Endpoints.CV.Dodaj;
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;

namespace JobSearchingWebApp.Endpoints.CV.Update
{
    public class CVUpdateRequest
    {
        public int Id { get; set; }

        public string KandidatId { get; set; }

        public string? Naziv { get; set; }

        public bool? Objavljen { get; set; }

        public string? Ime { get; set; }

        public string? Prezime { get; set; }

        //[AllowNull]
        //[EmailAddress]
        public string? Email { get; set; }

        public string? Drzava { get; set; }

        public string? Grad { get; set; }

        public string? ProfesionalniSazetak { get; set; }

        public List<string>? Vjestine { get; set; }

        public List<string>? TehnickeVjestine { get; set; }

        public List<string>? Kursevi { get; set; }

        public List<EdukacijaUpdateRequest>? Edukacija { get; set; }

        public List<ZaposlenjeUpdateRequest>? Zaposlenje { get; set; }

        public List<URLUpdateRequest>? URL { get; set; }
    }

    public class EdukacijaUpdateRequest
    {
        [Required]
        public int Id { get; set; }

        public string? NazivSkole { get; set; }

        public DateOnly? DatumPocetka { get; set; }

        public DateOnly? DatumZavrsetka { get; set; }

        public string? Grad { get; set; }

        public string? Opis { get; set; }
    }

    public class ZaposlenjeUpdateRequest
    {
        [Required]
        public int Id { get; set; }

        public string? NazivKompanije { get; set; }

        public string? NazivPozicije { get; set; }

        public DateOnly? DatumPocetka { get; set; }

        public DateOnly? DatumZavrsetka { get; set; }

        public string? Opis { get; set; }
    }

    public class URLUpdateRequest
    {
        [Required]
        public int Id { get; set; }

        public string? Naziv { get; set; }

        [Url]
        public string? Putanja { get; set; }
    }
}

File: front-end/src/app/endpoints/cv-endpoint/update/cv-update-request.ts
export class CvUpdateRequest {
  constructor(
    public id: number,
    public objavljen: boolean | null,
    public kandidatId: string | null,
    public naziv: string | null,
    public ime: string | null,
    public prezime: string | null,
    public email: string | null,
    public brojTelefona: string | null,
    public drzava: string | null,
    public grad: string | null,
    public profesionalniSazetak: string | null,
    public vjestine: string[] | null,
    public tehnickeVjestine: string[] | null,
    public kursevi: string[] | null,
    public edukacija: EdukacijaUpdateRequest[] | null,
    public zaposlenje: ZaposlenjeUpdateRequest[] | null,
    public url: UrlUpdateRequest[] | null,) {}
}

export interface EdukacijaUpdateRequest {
  id: number;
  nazivSkole: string | null;
  datumPocetka: Date | null;
  datumZavrsetka: Date | null;
  grad: string | null;
  opis: string | null;
}

export interface ZaposlenjeUpdateRequest {
  id: number;
  nazivKompanije: string | null;
  nazivPozicije: string | null;
  datumPocetka: string | null;
  datumZavrsetka: string | null;
  opis: string | null;
}

export interface UrlUpdateRequest {
  id: number;
  naziv: string | null;
  putanja: string | null;
}

